src/scipp/coords/transform_coords.py

```diff
@@ -14,14 +14,16 @@
 
 
 def transform_coords(x: Union[DataArray, Dataset],
-                     targets: Union[str, Iterable[str]],
-                     graph: GraphDict,
+                     targets: Optional[Union[str, Iterable[str], GraphDict]] = None,
+                     /,
+                     graph: Optional[GraphDict] = None,
                      *,
                      rename_dims: bool = True,
                      keep_aliases: bool = True,
                      keep_intermediate: bool = True,
                      keep_inputs: bool = True,
-                     quiet: bool = False) -> Union[DataArray, Dataset]:
+                     quiet: bool = False,
+                     **kwargs: Dict[str, Callable]) -> Union[DataArray, Dataset]:
```

Comment:
```suggestion
                     **kwargs: Callable) -> Union[DataArray, Dataset]:
```
`kwargs` are implicitly a dict with `str` keys and one only lists the value type.